include/oneapi/dpl/pstl/utils.h

```diff
@@ -70,19 +70,6 @@ __except_handler(_Fp __f) -> decltype(__f())
     }
 }
 
-template <typename _Op>
-struct __invoke_unary_op
-{
-    mutable _Op __op;
```

Comment:
Yeah, I notice that we lose `mutable`. This is the most interesting question of this review (expect prohibiting the assignment to rvalue fundamentals)

To me, we should do it consistent. Either both binary and unary transformations support `mutable` or neither. I don't see how unary transformation is special with that regard.

Reply:
I agree. 
When I was investigating the current state of oneDPL with regard to const-qualified `operator()` for function objects, this was the biggest inconsistency I found (binary vs unary overloads for `transform`).  I could not find any legitimate reason to justify it in the git history that I have access.  
In the long run, I would be in favor or dropping this and most other instances of `mutable` in oneDPL.

Reply:
Perhaps we should try to think of a way to "deprecate" this support for `mutable` in a way that can give the user a warning (if possible), and then eventually remove its support.

Reply:
This is probably a task for another PR, and perhaps once we have access to c++20 which should make something like this easier via concepts / constraints.  Also, this is likely only relevant to deprecate with the `dpcpp` backend.

Reply:
> Made both `mutable`, because we have such a test for unary transform. I don't want to occasionally introduce a breaking change.

May we think about do not mark this member as `mutable` but apply some logic
inside `void __transform_impl(_OutputType&& output, _Args&&... args) const` :
- if `_M_pred` has operator() declared as `const` - simple call it;
- otherwise create a local copy of `_M_pred` inside `__transform_impl` function and call `operator()` on it...

Reply:
This is an interesting idea, but we trade one issue for another.   This adds a requirement that the function object is CopyConstructible which I don't think is required by the standard.  Still, it's worth considering.